src/commandsAndMenu.tsx

```diff
@@ -368,168 +387,216 @@ export function addCommands(
-          nbDiffWidget.title.label = PathExt.basename(filePath);
-          nbDiffWidget.title.icon = diffIcon;
-          nbDiffWidget.title.closable = true;
-          nbDiffWidget.addClass('jp-git-diff-parent-diff-widget');
-
-          shell.add(nbDiffWidget, 'main');
-          shell.activateById(nbDiffWidget.id);
         }
-      } else {
-        showErrorMessage(
-          'Diff Not Supported',
-          `Diff is not supported for ${PathExt.extname(
-            filePath
-          ).toLocaleLowerCase()} files.`
-        );
       }
-    }
+    },
+    icon: diffIcon
   });
 
-  commands.addCommand(CommandIDs.gitFileStage, {
+  commands.addCommand(ContextCommandIDs.gitFileStage, {
     label: 'Stage',
-    caption: 'Stage the changes of selected file',
+    caption: pluralizedContextLabel(
+      'Stage the changes of selected file',
+      'Stage the changes of selected files'
+    ),
     execute: async args => {
-      const selectedFile: Git.IStatusFile = args as any;
-      await model.add(selectedFile.to);
-    }
+      const { files } = (args as any) as CommandArguments.IGitContextAction;
+      for (const file of files) {
+        await model.add(file.to);
+      }
+    },
+    icon: addIcon
   });
 
-  commands.addCommand(CommandIDs.gitFileTrack, {
+  commands.addCommand(ContextCommandIDs.gitFileTrack, {
     label: 'Track',
-    caption: 'Start tracking selected file',
+    caption: pluralizedContextLabel(
+      'Start tracking selected file',
+      'Start tracking selected files'
+    ),
     execute: async args => {
-      const selectedFile: Git.IStatusFile = args as any;
-      await model.add(selectedFile.to);
-    }
+      const { files } = (args as any) as CommandArguments.IGitContextAction;
+      for (const file of files) {
+        await model.add(file.to);
+      }
+    },
+    icon: addIcon
   });
 
-  commands.addCommand(CommandIDs.gitFileUnstage, {
+  commands.addCommand(ContextCommandIDs.gitFileUnstage, {
     label: 'Unstage',
-    caption: 'Unstage the changes of selected file',
+    caption: pluralizedContextLabel(
+      'Unstage the changes of selected file',
+      'Unstage the changes of selected files'
+    ),
     execute: async args => {
-      const selectedFile: Git.IStatusFile = args as any;
-      if (selectedFile.x !== 'D') {
-        await model.reset(selectedFile.to);
+      const { files } = (args as any) as CommandArguments.IGitContextAction;
+      for (const file of files) {
+        if (file.x !== 'D') {
+          await model.reset(file.to);
+        }
       }
-    }
+    },
+    icon: removeIcon
   });
 
-  commands.addCommand(CommandIDs.gitFileDelete, {
+  function representFiles(files: Git.IStatusFile[]): JSX.Element {
+    if (files.length > 1) {
+      const elements = files.map(file => (
+        <li key={file.to}>
+          <b>{file.to}</b>
+        </li>
+      ));
+      return <ul>{elements}</ul>;
+    } else {
+      return <b>{files[0].to}</b>;
```

Comment:
TODO: double check if there is a space between action and file name upon rendering (there was not prior to this PR for delete action).